### PR TITLE
[SB-1253] Fix Ogmigo crash

### PR DESCRIPTION
**Pull request by droark**
- When adding an asset to a Value, initialization may not have been performed. Perform proper outer map initialization.

### CODE REVIEW

**Comment by droark** [comment]
@Quantumplation - I tossed in the API change because it became clear while updating some code that the name should be changed. I figured I might as well fix that while fixing this crash, which affects unit tests and would obviously affect library usage.

**Comment by droark** [comment]
Removed an API change I made. It risks breaking code elsewhere in subtle ways. We should avoid that.

File: ouroboros/shared/value.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ type Coin struct {
 }
 
 func ValueFromCoins(coins ...Coin) Value {
-	var value Value
+	value := Value{}
```
Comment (malls) [inline ouroboros/shared/value.go:146]: I had to do a similar fix to get my branch to work. I'd appreciate this merged ASAP.

**Review by Quantumplation — APPROVED** [review]

File: ouroboros/shared/value.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ type Coin struct {
 }
 
 func ValueFromCoins(coins ...Coin) Value {
-	var value Value
-	value.AddAsset(coins...)
+	value := Value{}
```
[the remark this answers is not in the document]
Comment (Quantumplation) [reply ouroboros/shared/value.go:146]: oh nvm, because this is a map, so the zero value is nil

[the remark this answers is not in the document]
Comment (droark) [reply ouroboros/shared/value.go:146]: Right. Without this line, `AddAsset()` can crash at [these lines](https://github.com/SundaeSwap-finance/ogmigo/pull/28/files#diff-fdb3ffb22efc20ae342ed6283cfa9d78a10f3951cb0b2774d940f43696f5ed25R108-R110). With `value` properly initialized, everything's fine.

**Pull request merged**